### ISSUE TITLE
[v3] wrong signature generated for transaction

### ISSUE THREAD

**Issue by AyushBherwani1998** [Web3Auth/mpc-core-kit#148]
## Description 

Looks like some other account is being used to generate the signature for sending the transaction. I tried to decode the signature, and from address is different, which results in insufficient funds in account error.

- From Address(default selected index - 0): `0x7d5F4C52D2bA65B148d973dd1F8503fF5Fe3eADe`

- Generated Signature: 
```
0x02f8728189808506fc23ac0085072a3c6fd282520894dc5afb5de928bac740e4bcb1600b93504560d85085e8d4a5100080c001a071b9e00d3c97f433a7b685606f9b6bccc1ed1ee283362921003f51931a78b2ffa04f55c56a408444ea7e64f4f0abdc0a57aceea8d854080c44485f89812772af93
```

- Decoded Signature: 
```
{
  "type": 2,
  "chainId": 137,
  "nonce": 0,
  "maxPriorityFeePerGas": {
    "type": "BigNumber",
    "hex": "0x06fc23ac00"
  },
  "maxFeePerGas": {
    "type": "BigNumber",
    "hex": "0x072a3c6fd2"
  },
  "gasPrice": null,
  "gasLimit": {
    "type": "BigNumber",
    "hex": "0x5208"
  },
  "to": "0xDC5aFb5DE928bAc740e4bcB1600B93504560d850",
  "value": {
    "type": "BigNumber",
    "hex": "0xe8d4a51000"
  },
  "data": "0x",
  "accessList": [],
  "hash": "0x7c45a8400cad4b2034ab10849666f56c9aa16b63ab958e5c6c1a00e1c49776c1",
  "v": 1,
  "r": "0x71b9e00d3c97f433a7b685606f9b6bccc1ed1ee283362921003f51931a78b2ff",
  "s": "0x4f55c56a408444ea7e64f4f0abdc0a57aceea8d854080c44485f89812772af93",
  "from": "0x7dF92DF5760D058eB3FfcE598678D016104D5113"
}
```

Check the `from`, it's not the same as the From address which results in insufficient funds from the RPC . 

## Reproducible Steps:

1. Use the v3 branch demo
2. In `sendTransaction` method add  `signTransaction` to generate signature for the transaction as well.
```ts
try {
      const receipt = await web3.eth.signTransaction({
        from: fromAddress,
        to: destination,
        value: amount,
      });
      uiConsole(receipt);
    } catch (e) {
      uiConsole(e)
    }
```
3. Switch to Polygon mainnet
4. Try to sendTrasnaction and generate the signature
5. Decode the signature

**Comment by matthiasgeihs** [comment]
The issue was that `makeEthereumSigner` did not specify that the input was a hash.

Pushed a fix to https://github.com/Web3Auth/mpc-core-kit/pull/147.

**Comment by AyushBherwani1998** [comment]
Works fine in v3-alpha

**Comment by matthiasgeihs** [comment]
Resolved in #147
